File: temp/fix7.js
const fs = require('fs');
const path = require('path');

const slidesDir = 'D:/新课开发/工作手册/创新路径-突破口-工作手册/完整课程包/03-授课PPT/slides';

const skipFiles = ['node_modules', 'compile.js'];

// Fix inner quotes in strings
// Strategy: For lines containing addText, find the string boundaries and escape inner quotes

function fixInnerQuotes(line) {
  // Only process lines with addText
  if (!line.includes('addText')) return line;
  
  // Find the opening and closing quotes for the first string argument
  // Pattern: addText("..." or addText("...", {
  // We need to find the first " after addText( and the matching closing "
  
  const result = [];
  let i = 0;
  let inString = false;
  let stringStart = -1;
  
  while (i < line.length) {
    const char = line[i];
    
    if (!inString) {
      result.push(char);
      if (char === '"') {
        inString = true;
        stringStart = result.length - 1;
      }
      i++;
    } else {
      if (char === '\') {
        // Escape sequence - keep both characters
        result.push(char);
        i++;
        if (i < line.length) {
          result.push(line[i]);
          i++;
        }
      } else if (char === '"') {
        // Check if this is the end of the string or an inner quote
        // Look ahead to see what's next
        const nextChar = line[i + 1];
        if (nextChar === '"' || nextChar === ',' || nextChar === '\n' || nextChar === ')') {
          // This is the end of the string
          result.push(char);
          inString = false;
          i++;
        } else {
          // This is an inner quote - escape it
          result.push('\\"');
          i++;
        }
      } else {
        result.push(char);
        i++;
      }
    }
  }
  
  return result.join('');
}

// Test on a specific problematic file
const slide08 = path.join(slidesDir, 'slide-08.js');
let content = fs.readFileSync(slide08, 'utf8');
const lines = content.split('\n');

let fixed = false;
const newLines = lines.map(line => {
  if (line.includes('addText') && line.includes('"知道"')) {
    const fixedLine = fixInnerQuotes(line);
    if (fixedLine !== line) {
      fixed = true;
      return fixedLine;
    }
  }
  return line;
});

if (fixed) {
  fs.writeFileSync(slide08, newLines.join('\n'));
  console.log('Fixed slide-08');
} else {
  console.log('No fix needed for slide-08');
}

// Check what the problematic line looks like
const line = lines[84];
console.log('Line 85:', line);
console.log('Has inner quotes:', line.includes('"知道"') || line.includes('"做到"'));

File: temp/fix7.py
#!/usr/bin/env python3
"""Fix 7 broken slides by proper string parsing."""
import re, os

SLIDES_DIR = r"D:\新课开发\内训师和表达\系列进阶课\12-内训项目运营：选师、建课、认证、应用与激励机制设计\授课PPT\slides"
FILES = ['slide-36.js','slide-49.js','slide-67.js','slide-76.js','slide-80.js','slide-97.js','slide-98.js']

def is_cjk(ch):
    return '一' <= ch <= '鿿'

def parse_and_fix(line):
    """Parse a JS line, identify string values, fix inner quotes."""
    result = []
    i = 0
    in_string = False
    string_char = None

    while i < len(line):
        ch = line[i]

        if not in_string:
            if ch in ('"', "'"):
                in_string = True
                string_char = ch
                result.append(ch)
                i += 1
            elif ch == '/' and i + 1 < len(line) and line[i + 1] == '/':
                # Comment - rest of line
                result.append(line[i:])
                break
            else:
                result.append(ch)
                i += 1
        else:
            if ch == '\\' and i + 1 < len(line) and line[i + 1] in ('"', "'", '\\', 'n', 't', 'r'):
                # Escaped char - keep as is
                result.append(ch)
                result.append(line[i + 1])
                i += 2
            elif ch == string_char:
                # End of string
                in_string = False
                string_char = None
                result.append(ch)
                i += 1
            elif ch == '"' and string_char == '"':
                # Unexcaped double quote inside double-quoted string = inner quote!
                # Check if it's between CJK chars
                prev_c = line[i - 1] if i > 0 else ''
                next_c = line[i + 1] if i + 1 < len(line) else ''
                if is_cjk(prev_c) or is_cjk(next_c):
                    # Inner quote - escape it
                    result.append('\\"')
                    i += 1
                else:
                    # Something else
                    result.append('\\"')
                    i += 1
            else:
                result.append(ch)
                i += 1

    return ''.join(result)

for fname in FILES:
    fpath = os.path.join(SLIDES_DIR, fname)
    if not os.path.exists(fpath):
        print(f'NOT FOUND: {fpath}')
        continue

    with open(fpath, 'r', encoding='utf-8') as f:
        lines = f.readlines()

    new_lines = []
    changed = False
    for line in lines:
        stripped = line.rstrip('\r\n')
        if '"' in stripped and not stripped.startswith('//'):
            new_stripped = parse_and_fix(stripped)
            if new_stripped != stripped:
                line = new_stripped + '\n'
                changed = True
        new_lines.append(line)

    if changed:
        with open(fpath, 'w', encoding='utf-8') as f:
            f.writelines(new_lines)
        print(f'Fixed: {fname}')

print('Done')
